### PR TITLE
MM-60312- rename ssoOffice 365 to sso EntraId

### PR DESCRIPTION
**Pull request by pvev**
#### Summary
rename ssoOffice 365 to sso EntraId

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-60312

#### Checklist

- [x ] Added or updated unit tests (required for all new features)
- [x ] Has UI changes
- [ x] Includes text changes and localization file updates
- [ ] Have tested against the 5 core themes to ensure consistency between them.
- [ ] Have run E2E tests by adding label `E2E iOS tests for PR`.

#### Device Information
This PR was tested on: Androis and IOS

#### Screenshots
Screenshot 2024-09-06 at 13 34 26
Screenshot 2024-09-06 at 14 23 07

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Comment by enahum** [comment]
Tests failing

File: app/screens/edit_profile/components/email_field.tsx [hunk 1]
```diff
@@ -15,7 +15,7 @@ import type {FloatingTextInputRef} from '@components/floating_text_input_label';
 const services: Record<string, string> = {
     gitlab: 'GitLab',
     google: 'Google Apps',
-    office365: 'Office 365',
+    office365: 'Entra ID',
```
Comment (enahum) [inline app/screens/edit_profile/components/email_field.tsx:18]: Why not change the prop name as well?

File: app/screens/login/sso_options.tsx [hunk 2]
```diff
@@ -46,8 +46,8 @@ const SsoOptions = ({goToSso, ssoOnly, ssoOptions, theme}: Props) => {
                 sso.imageSrc = require('@assets/images/Icon_Google.png');
                 break;
             case Sso.OFFICE365:
-                sso.text = formatMessage({id: 'mobile.login_options.office365', defaultMessage: 'Office 365'});
-                sso.imageSrc = require('@assets/images/Icon_Office.png');
```
Comment (enahum) [inline app/screens/login/sso_options.tsx:50]: File should be deleted

File: app/screens/login/sso_options.tsx [hunk 3]
```diff
@@ -46,8 +46,8 @@ const SsoOptions = ({goToSso, ssoOnly, ssoOptions, theme}: Props) => {
...
+                sso.text = formatMessage({id: 'mobile.login_options.office3651', defaultMessage: 'Entra ID'});
```
Comment (enahum) [inline app/screens/login/sso_options.tsx:49]: Id should change to entraid

File: assets/base/i18n/en.json [hunk 4]
```diff
@@ -661,7 +661,7 @@
   "mobile.login_options.google": "Google",
   "mobile.login_options.heading": "Log In to Your Account",
   "mobile.login_options.none": "You can't log in to your account yet. At least one login option must be configured. Contact your System Admin for assistance.",
-  "mobile.login_options.office365": "Office 365",
+  "mobile.login_options.office365": "Entra ID",
```
Comment (enahum) [inline assets/base/i18n/en.json:664]: Id should match the change mentioned in the other comment

**Review by enahum — CHANGES_REQUESTED** [review]

File: app/screens/edit_profile/components/email_field.tsx (hunk 1 above)
Comment (pvev) [reply app/screens/edit_profile/components/email_field.tsx:18]: the agreement was to keep all the ids, props, keys, etc to keep referencing as office365. Same happened in webapp and in server.

File: app/screens/login/sso_options.tsx (hunk 2 above)
Comment (pvev) [reply app/screens/login/sso_options.tsx:50]: thanks for confirming this, I was not sure and forgot to ask 👍

File: app/screens/login/sso_options.tsx (hunk 3 above)
Comment (pvev) [reply app/screens/login/sso_options.tsx:49]: ok

File: assets/base/i18n/en.json (hunk 4 above)
Comment (pvev) [reply assets/base/i18n/en.json:664]: ok

**Review by larkox — APPROVED** [review]

**Review by enahum — APPROVED** [review]

**Review by lindalumitchell — APPROVED** [review]
LGTM; no issues found. Verified login button, text in profile details.

**Pull request merged**